src/applications/ivc-champva/10-10D/components/CustomRelationshipPattern.jsx

```diff
@@ -0,0 +1,145 @@
+import {
+  radioUI,
+  radioSchema,
+} from 'platform/forms-system/src/js/web-component-patterns';
+import VaTextInputField from 'platform/forms-system/src/js/web-component-fields/VaTextInputField';
+
+/**
+ * Customized web component uiSchema for relationship to veteran
+ * with support for more custom wording.
+ *
+ * set `labelHeaderLevel: ''` if you just want to use it as a standard field instead
+ *
+ * ```js
+ * relationshipToVeteran: relationshipToVeteran() // 'Veteran'
+ * relationshipAsField: relationshipToVeteran({
+ *  personTitle: 'claimant',
+ *  labelHeaderLevel: ''
+ * })
+ * customRelationshipToVeteran: relationshipToVeteranUI({
+ *   customLabels: {
+ *     caretaker: "I’m the Veteran’s caretaker"
+ *   }
+ * })
+ * ```
+ *
+ * @param {string | {
+ *   personTitle?: string,
+ *   relativeTitle?: string,
+ *   tense?: string,
+ *   labelHeaderLevel?: UISchemaOptions['ui:options']['labelHeaderLevel'],
+ *   customLabels?: object
+ * }} options - a string representing the person type, or an object with options
+ * @returns {UISchemaOptions}
+ */
+
+export const relationshipToVeteranUI = options => {
+  const { personTitle, relativeTitle, tense, labelHeaderLevel, customLabels } =
+    typeof options === 'object' ? options : { personTitle: options };
+
+  const person = personTitle ?? 'Veteran';
+
+  const isRelativeFn = typeof relativeTitle === 'function';
+  const relativeTitleVal = isRelativeFn
+    ? props => relativeTitle(props)
+    : relativeTitle;
+
+  const conjugatedVerbThirdPerson = tense && tense === 'past' ? 'was' : 'is';
+  const conjugatedVerbFirstPerson = tense && tense === 'past' ? 'I was' : 'I’m';
+
+  const relativeBeingVerb = `${`${
+    relativeTitleVal
+      ? `${relativeTitleVal} ${conjugatedVerbThirdPerson}`
+      : conjugatedVerbFirstPerson
+  }`}`;
+
+  const relativePossessive = `${`${
+    relativeTitleVal ? `${relativeTitleVal}’s` : 'your'
+  }`}`;
+
+  return {
+    relationshipToVeteran: radioUI({
+      title: `What’s ${relativePossessive} relationship to the ${person}?`,
+      labels: customLabels || {
+        spouse: `${relativeBeingVerb} the ${person}’s spouse`,
+        child: `${relativeBeingVerb} the ${person}’s child`,
+        caretaker: `${relativeBeingVerb} the ${person}’s caretaker`,
+        other: `${`${
+          relativeTitleVal ? `${relativeTitleVal} doesn’t` : 'We don’t'
+        }`} have a relationship that’s listed here`,
```

Comment:
Can this be refactored into something similar to what was done here - https://github.com/department-of-veterans-affairs/vets-website/pull/28028#discussion_r1493022109

Reply:
Ah, thanks for catching that! Yes, I've adjusted it to have the same solution. Also for context, the goal is to have this relationship component replace the existing one referenced there.